x-pack/plugins/fleet/server/routes/data_streams/handlers.ts

```diff
@@ -51,10 +56,22 @@ export const getListHandler: RequestHandler = async (context, request, response)
   };
 
   try {
+    const useMeteringApi = appContextService.getConfig()?.internal?.useMeteringApi;
+
     // Get matching data streams, their stats, and package SOs
-    const [dataStreamsInfo, dataStreamStats, packageSavedObjects] = await Promise.all([
+    const [
+      dataStreamsInfo,
+      dataStreamStatsOrUndefined,
+      dataStreamMeteringStatsorUndefined,
+      packageSavedObjects,
+    ] = await Promise.all([
       dataStreamService.getAllFleetDataStreams(esClient),
-      dataStreamService.getAllFleetDataStreamsStats(esClient),
+      useMeteringApi
+        ? undefined
+        : dataStreamService.getAllFleetDataStreamsStats(elasticsearch.client.asSecondaryAuthUser),
+      useMeteringApi
+        ? dataStreamService.getAllFleetMeteringStats(elasticsearch.client.asSecondaryAuthUser)
```

Comment:
why use `asSecondaryAuthUser` here?

Reply:
The API is only available as internal, and calling it as current user fail for that reason.